astropy/time/formats.py

```diff
@@ -1187,7 +1220,38 @@ class TimeString(TimeUnique):
     This class assumes that anything following the last decimal point to the
     right is a fraction of a second.
 
-    This is a reference implementation can be made much faster with effort.
+    **Fast C-based parser**
+
+    Time format classes can take advantage of a fast C-based parser if the times
+    are represented as fixed-format strings with year, month, day-of-month,
+    hour, minute, second, OR year, day-of-year, hour, minute, second. This can
+    be a factor of 20 or more faster than the pure Python parser.
+
+    Fixed format means that the components always have the same number of
+    characters. The Python parser will accept ``2001-9-2`` as a date, but the C
+    parser would require ``2001-09-02``.
+
+    A subclass in this case must define a class attribute ``fast_parser_pars``
+    which is a `dict` with all of the keys below. An inherited attribute is not
+    checked, only an attribute in the class ``__dict__``.
+
+    - ``delims`` (tuple of int): ASCII code for character at corresponding
+      ``starts`` position (0 => no character)
+
+    - ``starts`` (tuple of int): position where component starts (including
+      delimiter if present). Use -1 for the month component for format that use
+      day of year.
+
+    - ``stops`` (tuple of int): position where component ends. Use -1 to
+      continue to end of string, or for the month component for formats that use
+      day of year.
+
+    - ``break_allowed`` (tuple of int): if true (1) then the time string can
+          legally end just before the corresponding component (e.g. "2000-01-01"
+          is a valid time but "2000-01-01 12" is not).
+
+    - ``has_day_of_year`` (int): 0 if dates have year, month, day; 1 if year,
```

Comment:
Could one just infer this based on the month being absent?

Reply:
Maybe. OTOH explicit is better than implicit. Leaving this.